src/main/java/xyz/vegaone/scribe/service/FileReaderServiceImpl.java

```diff
@@ -1,24 +1,64 @@
 package xyz.vegaone.scribe.service;
 
-import java.util.List;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.pdfbox.pdmodel.PDDocument;
+import org.apache.pdfbox.text.PDFTextStripper;
+import org.springframework.stereotype.Service;
 
-public class FileReaderServiceImpl implements FileReaderService{
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
 
-    public static void main(String[] args) {
+@Slf4j
+@Service
+public class FileReaderServiceImpl implements FileReaderService {
 
-    }
     @Override
-    public List<String> contentReaderPdf() {
-        return null;
+    public Map<Integer, String> contentReaderPdf() throws IOException {
+
+        Map<Integer, String> pageNumberAndText = new HashMap<>();
+
+        List<String> textFromFile = new ArrayList<>();
+
+        FileDiscoveryServiceImpl fileDiscoveryService = new FileDiscoveryServiceImpl();
+        List<String> name = fileDiscoveryService.discoverFiles("C:\\Test").get("pdf");
```

Comment:
We want to decouple the `contentReaderPdf()` method from `fileDiscoveryService`. The path to the file should be passed through a parameter. The method should have a single purpose/responsibility, that is to parse a file.
Ping me when you'll address this issue, I have an idea that we could apply here.